### PR TITLE
#1 Added support for title color in alternate manner

### PR DESCRIPTION
**Pull request by noahCruncher**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,87 +1,20 @@
...
+import 'domain/Song.dart';
 
 Future<void> main() async {
   WidgetsFlutterBinding.ensureInitialized();
-  List<Map> songs = await DatabaseHandler().retrieveAttributes('SELECT title, comments FROM songs', 'title');
-  runApp(MyApp(
-    songs: List<Map>.generate(songs.length, (index) => songs[index]),
+  List<Song> songs = await DatabaseHandler().findAllSongs();
+  List<Author> authors = await DatabaseHandler().findAllAuthors();
+  List<Map> books = await DatabaseHandler().retrieveAttributes('select sb.name, count(ssb.song_id) as songs from song_books sb, songs_songbooks ssb where sb.id = ssb.songbook_id group by sb.name', 'name');
```
Comment (VigneshPalanisamy) [inline lib/main.dart:12]: can we create a domain in name of song_book with only name and id. and later in that view we will execute and get the list of songs from it by song_book ID.

File: lib/main.dart [hunk 2]
```diff
@@ -1,87 +1,20 @@
...
+  List<Map> topics = await DatabaseHandler().retrieveAttributes('select t.name, count(st.song_id) as songs from topics t, songs_topics st where t.id = st.topic_id group by t.name', 'name');
```
Comment (VigneshPalanisamy) [inline lib/main.dart:13]: can we create a domain in name of topics with only name and id. and later in that view we will execute and get the list of songs from it by song_topics ID.

File: test/widget_test.dart [hunk 3]
```diff
@@ -5,26 +5,28 @@
...
-    expect(find.text('1'), findsNothing);
+  List<Song> songs = [Song(id: 1, title: "Seekiramaai Vanthiduvaen Endru Sonneerae", lyrics: "test lyrics")];
+  List<Author> authors = [Author(id: 6, firstName: "Fr. S. J.", lastName: "Berchmans", displayName: "Fr. S. J. Berchmans {பெர்க்மான்ஸ்}")];
 
-    // Tap the '+' icon and trigger a frame.
-    await tester.tap(find.byIcon(Icons.add));
-    await tester.pump();
+  test('Check the Song count from a map', ()
+  {
+    expect(songs.length, 1);
```
Comment (VigneshPalanisamy) [inline test/widget_test.dart:20]: why we need this test

File: pubspec.yaml [hunk 4]
```diff
@@ -34,7 +34,7 @@ dependencies:
 
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
-  cupertino_icons: ^1.0.6
+  cupertino_icons:
   sqflite:
```
Comment (VigneshPalanisamy) [inline pubspec.yaml:38]: can we specify the versions for the dependencies.

File: lib/domain/Song.dart [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  String lyrics = "";
+  String? verseOrder = "";
+  String? lastModified;
+  String? comment  = "";
+
+
+  Song({required this.id, required this.title, this.alternateTitle, required this.lyrics, this.verseOrder,
+      this.lastModified, this.comment});
+
+  factory Song.fromMap(Map<String, dynamic> json) => Song(
```
Comment (VigneshPalanisamy) [inline lib/domain/Song.dart:14]: write test for domains

File: lib/domain/Song.dart [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class Song {
+  int id;
+  String title = "";
+  String? alternateTitle = "";
+  String lyrics = "";
```
Comment (VigneshPalanisamy) [inline lib/domain/Song.dart:5]: keep the required fields in top

File: lib/db/DatabaseHandler.dart [hunk 7]
```diff
@@ -47,45 +50,28 @@ class DatabaseHandler{
...
+    var response = await database.query("songs");
+    List<Song> list = response.map((c) => Song.fromMap(c)).toList();
+    return list;
   }
 
-  @override
-  String toString() {
-    return 'Song{id: $id, name: $title}';
+  Future<List<Author>> findAllAuthors() async {
+    Database database = await initializedDB();
```
Comment (VigneshPalanisamy) [inline lib/db/DatabaseHandler.dart:72]: do we need to initialize the Db in every method? shall we initialize in init

File: lib/db/DatabaseHandler.dart [hunk 8]
```diff
@@ -47,45 +50,28 @@ class DatabaseHandler{
     return await openDatabase(path, readOnly: true);
   }
 
-  Future<List<Map>> retrieveAttributes(String query, String attribute) async {
+  Future<List<Map>> getSongs() async {
```
Comment (VigneshPalanisamy) [inline lib/db/DatabaseHandler.dart:53]: do we need this

**Review by VigneshPalanisamy — COMMENTED** [review]
given some comments

File: pubspec.yaml (hunk 4 above)
Comment (noahCruncher) [reply pubspec.yaml:38]: Is it really need to specify the version?

File: lib/main.dart (hunk 2 above)
Comment (noahCruncher) [reply lib/main.dart]: Removed this for now. When we add the topics then we will add it.

File: lib/main.dart (hunk 1 above)
Comment (noahCruncher) [reply lib/main.dart]: Removed this for now. When we add the books then we will add it.

**Review by VigneshPalanisamy — APPROVED** [review]

**Pull request merged**